bibangamba/GitHubUsers
# feat(mvp-refactor): refactor project to mvp architecture pattern

#### What does this PR do?
refactors the project to the MVP architecture pattern

#### Description of tasks to be done?
- change the project structure to follow [Google MVP blueprint](https://github.com/googlesamples/android-architecture/tree/todo-mvp)
- ensure that all the tests still pass

#### How can this be manually tested?
- clone the repo and check out the branch
- compare the project structure to what is showcased in the MVP blueprint above
- the architecture and flow of data should be very similar
- run the tests using `./gradlew jacocoTestReport`
- all the tests should pass

#### What are the relevant pivotal tracker stories?
[#158790835](https://www.pivotaltracker.com/story/show/158790835)
